### DISCUSSION TITLE
How to load custom models

### DISCUSSION THREAD

**Discussion by qq527364584** [open-webui/open-webui#4140] [General]
How to load the model you fine-tuned and trained yourself into the open webui。

**Comment by GylanSalih** [comment]
Here is a articel, i hope that can help you buddy:

> Save Your Fine-Tuned Model: Ensure your model is saved in a format compatible with the Open WebUI. This might be a specific file format or directory structure depending on the model and framework you used for fine-tuning.
> 
> Prepare the Model for Deployment: If you used a framework like Hugging Face’s Transformers, PyTorch, or TensorFlow, make sure your model is exported correctly. For example, in Hugging Face Transformers, you’d save the model with model.save_pretrained('path_to_save') and the tokenizer with tokenizer.save_pretrained('path_to_save').
> 
> Access Open WebUI’s Model Management: Open WebUI should have an interface or configuration file where you can specify which model to use. This is usually done via a settings menu or a configuration file.
> 
> Upload the Model: If Open WebUI provides a way to upload models directly through its interface, use that method to upload your fine-tuned model. This might involve specifying the path to the model directory or uploading the files via a web interface.
> 
> Configure Model Settings: Set any necessary configuration options in Open WebUI to ensure that it correctly interacts with your fine-tuned model. This might involve setting parameters for loading the model, specifying its type, or adjusting input/output formats.
> 
> Restart or Refresh: After uploading and configuring your model, you might need to restart the application or refresh the interface to load the new model.
> 
> Test the Integration: Verify that the model is working as expected within Open WebUI. Test it with sample inputs to ensure that it performs well and integrates seamlessly with the interface.
